libcni/api.go

```diff
@@ -175,6 +178,16 @@ func (c *CNIConfig) ensureExec() invoke.Exec {
 	return c.exec
 }
 
+type cachedInfo struct {
+	Kind           string                 `json:"kind"`
+	CNIVersion     string                 `json:"cniVersion"`
+	Config         []byte                 `json:"config"`
+	CniArgs        string                 `json:"cniArgs,omitempty"`
```

Comment:
@mccv1r0 If we're going full JSON here we might as well make CniArgs just be a `[][2]string` just like it is in RuntimeConf and then we don't need stringify/unstringify.